src/main.rs

```diff
@@ -25,11 +39,248 @@ fn main() {
+        let input_slice: Vec<&[f32]> = input.iter().map(|row| row.as_slice()).collect();
+        let mut output_slice: Vec<&mut [f32]> = output.iter_mut().map(|row| row.as_mut_slice()).collect();
+
+        comb_filter.process(input_slice.as_slice(), output_slice.as_mut_slice());
+       
+        for t in 100..num_samples {
+            assert!(output[t][0].abs() < 1.0e-7);
+        }
+    }
+
+    #[test]
+    fn fir_constant_non_zero_input() {
+        let filter_type = comb_filter::FilterType::FIR;
+        let sample_rate = 8000 as u32;
+        let channels = 1 as u16;
+
+        let gain = 0.5;
+        let delay = 0.01;
+
+        let mut comb_filter = comb_filter::CombFilter::new(filter_type, delay, sample_rate as f32, channels as usize);
+        comb_filter.set_param(comb_filter::FilterParam::Gain, gain);
+
+        let duration_secs = 3.0;
+        
+        // Calculate the number of samples
+        let num_samples = (sample_rate as f32 * duration_secs) as usize;
+        let mut input = vec![vec![0.5 as f32]; num_samples];
+
+        let mut output = vec![vec![0.0 as f32]; num_samples];
+        let input_slice: Vec<&[f32]> = input.iter().map(|row| row.as_slice()).collect();
+        let mut output_slice: Vec<&mut [f32]> = output.iter_mut().map(|row| row.as_mut_slice()).collect();
+
+        comb_filter.process(input_slice.as_slice(), output_slice.as_mut_slice());
+       
+        for t in 0..80 {
+            assert!((output[t][0] - 0.5).abs() < 1.0e-7);
+        }
+        for t in 80..num_samples {
+            assert!((output[t][0] - 0.75).abs() < 1.0e-7);
+        }
+    }
+
+    #[test]
+    fn iir_constant_non_zero_input() {
+        let filter_type = comb_filter::FilterType::IIR;
+        let sample_rate = 8000 as u32;
+        let channels = 1 as u16;
+
+        let gain = 0.5;
+        let delay = 0.01;
+
+        let mut comb_filter = comb_filter::CombFilter::new(filter_type, delay, sample_rate as f32, channels as usize);
+        comb_filter.set_param(comb_filter::FilterParam::Gain, gain);
+
+        let duration_secs = 3.0;
+        
+        // Calculate the number of samples
+        let num_samples = (sample_rate as f32 * duration_secs) as usize;
+        let mut input = vec![vec![0.5 as f32]; num_samples];
+
+        let mut output = vec![vec![0.0 as f32]; num_samples];
+        let input_slice: Vec<&[f32]> = input.iter().map(|row| row.as_slice()).collect();
+        let mut output_slice: Vec<&mut [f32]> = output.iter_mut().map(|row| row.as_mut_slice()).collect();
+
+        comb_filter.process(input_slice.as_slice(), output_slice.as_mut_slice());
+       
+        for t in 0..80 {
+            assert!((output[t][0] - 0.5).abs() < 1.0e-7);
+        }
+        for t in 80..160 {
+            assert!((output[t][0] - 0.75).abs() < 1.0e-7);
+        }
+        for t in 160..240 {
+            assert!((output[t][0] - 0.875).abs() < 1.0e-7);
+        }
+    }
+
+}
```

Comment:
I would consider looking into implementing the blocks which I'm working on doing myself and adding some tests for `reset()`.